cmd/mt-whisper-importer-writer/main.go

```diff
@@ -87,112 +59,100 @@ func init() {
-	globalFlags.BoolVar(&storeConfig.CreateKeyspace, "cassandra-create-keyspace", storeConfig.CreateKeyspace, "enable the creation of the mdata keyspace and tables, only one node needs this")
-	globalFlags.BoolVar(&storeConfig.DisableInitialHostLookup, "cassandra-disable-initial-host-lookup", storeConfig.DisableInitialHostLookup, "instruct the driver to not attempt to get host info from the system.peers table")
-	globalFlags.BoolVar(&storeConfig.SSL, "cassandra-ssl", storeConfig.SSL, "enable SSL connection to cassandra")
-	globalFlags.StringVar(&storeConfig.CaPath, "cassandra-ca-path", storeConfig.CaPath, "cassandra CA certificate path when using SSL")
-	globalFlags.BoolVar(&storeConfig.HostVerification, "cassandra-host-verification", storeConfig.HostVerification, "host (hostname and server cert) verification when using SSL")
-	globalFlags.BoolVar(&storeConfig.Auth, "cassandra-auth", storeConfig.Auth, "enable cassandra authentication")
-	globalFlags.StringVar(&storeConfig.Username, "cassandra-username", storeConfig.Username, "username for authentication")
-	globalFlags.StringVar(&storeConfig.Password, "cassandra-password", storeConfig.Password, "password for authentication")
-	globalFlags.StringVar(&storeConfig.SchemaFile, "cassandra-schema-file", storeConfig.SchemaFile, "File containing the needed schemas in case database needs initializing")
-
-	cassFlags := cassandra.ConfigSetup()
-
-	flag.Usage = func() {
-		fmt.Println("mt-whisper-importer-writer")
-		fmt.Println()
-		fmt.Println("Opens an endpoint to send data to, which then gets stored in the MT internal DB(s)")
-		fmt.Println()
-		fmt.Printf("Usage:\n\n")
-		fmt.Printf("  mt-whisper-importer-writer [global config flags] <idxtype> [idx config flags] \n\n")
-		fmt.Printf("global config flags:\n\n")
-		globalFlags.PrintDefaults()
-		fmt.Println()
-		fmt.Printf("idxtype: only 'cass' supported for now\n\n")
-		fmt.Printf("cass config flags:\n\n")
-		cassFlags.PrintDefaults()
-		fmt.Println()
-		fmt.Println("EXAMPLES:")
-		fmt.Println("mt-whisper-importer-writer -cassandra-addrs=192.168.0.1 -cassandra-keyspace=mydata -exit-on-error=true -fake-avg-aggregates=true -http-endpoint=0.0.0.0:8080 -num-partitions=8 -partition-scheme=bySeries -ttls=8d,2y -uri-path=/chunks -verbose=true -cassandra-window-factor=20 cass -hosts=192.168.0.1:9042 -keyspace=mydata")
-	}
+	flag.Parse()
 
-	if len(os.Args) == 2 && (os.Args[1] == "-h" || os.Args[1] == "--help") {
-		flag.Usage()
-		os.Exit(0)
+	// Only try and parse the conf file if it exists
+	path := ""
+	if _, err := os.Stat(*confFile); err == nil {
+		path = *confFile
 	}
-
-	var cassI int
-	for i, v := range os.Args {
-		if v == "cass" {
-			cassI = i
-		}
-	}
-	if cassI == 0 {
-		fmt.Println("only indextype 'cass' supported")
-		flag.Usage()
+	config, err := globalconf.NewWithOptions(&globalconf.Options{
+		Filename:  path,
+		EnvPrefix: "IMPORTER_",
```

Comment:
We'll probably want to be able to specify overrides which only apply to the importer, and not the MTs. That's the the main reason why I gave it a different prefix. 
That way HM-API can first generate all the `MT_` env vars as it would for an MT write deployment, search&replace `MT_` with `IMPORTER_` and then apply the `IMPORTER_` overrides.

Reply:
wouldn't all the settings between importer and MT pods be largely the same? (same index and store configuration for example).
and if there are differences, well, they run in different pods, so they can't clobber one another.
using a different prefix seems to create more work for no reason that i can see (even you mention it would need custom search/replace code)

Reply:
mostly the settings will be the same. but we might want to adjust stuff, like for example a different write concurrency or write queue size etc, so i just didn't want to make that impossible

Reply:
again, different pods, so how would applying a different setting be impossible when using the same prefix?

Reply:
> so how would applying a different setting be impossible when using the same prefix?

Because we only have one section for overrides in the whole instance config.

Alternatively, I could make the writer parse the variables with `MT_`, but in HM-API I allow the user to add overrides prefixed with `IMPORTER_` which would then be searched & replaced to `MT_` before the env vars get added to the importer deployment. Those two possible solutions seem about equally good to me.

Reply:
aha so the clobbering is in the configuration itself, not in the application of it. makes sense.

All of the MT programs currently use the `MT_` prefix. 

```
grep -r EnvPrefix: cmd
cmd/mt-store-cat/main.go:		EnvPrefix: "MT_",
cmd/mt-kafka-mdm-sniff-out-of-order/main.go:		EnvPrefix: "MT_",
cmd/metrictank/metrictank.go:		EnvPrefix: "MT_",
cmd/mt-whisper-importer-writer/main.go:		EnvPrefix: "IMPORTER_",
cmd/mt-kafka-mdm-sniff/main.go:		EnvPrefix: "MT_",
```

I have a slight preference for consistency here (all using `MT_`) because it allows for easier config management and is simpler to work with.
The fact that hosted-metrics-api currently doesn't have a good way to specify different configuration for different tools or pods, is a problem that we can solve easily in hosted-metrics-api, and as you also mentioned, so let's do that.

Reply:
ok, changing it to `MT_`